community/io/src/main/java/org/neo4j/io/pagecache/PageCursor.java

```diff
@@ -297,23 +293,49 @@
      * Discern whether an out-of-bounds access has occurred since the last call to {@link #next()} or
      * {@link #next(long)}, or since the last call to {@link #shouldRetry()} that returned {@code true}, or since the
      * last call to this method.
+     *
      * @return {@code true} if an access was out of bounds, or the {@link #raiseOutOfBounds()} method has been called.
      */
     public abstract boolean checkAndClearBoundsFlag();
 
     /**
+     * Check if a cursor error has been set on this or any linked cursor, and if so, remove it from the cursor
+     * and throw it as a {@link CursorException}.
+     */
+    public abstract void checkAndClearCursorError() throws CursorException;
+
+    /**
      * Explicitly raise the out-of-bounds flag.
+     *
      * @see #checkAndClearBoundsFlag()
      */
     public abstract void raiseOutOfBounds();
 
     /**
+     * Set an error condition on the cursor with the given message.
+     * <p>
+     * This will make calls to {@link #checkAndClearCursorError()} throw a {@link CursorException} with the given
+     * message, unless the error has gotten cleared by a {@link #shouldRetry()} call that returned {@code true},
+     * a call to {@link #next()} or {@link #next(long)}, or the cursor is closed.
+     *
+     * @param message The message of the {@link CursorException} that {@link #checkAndClearCursorError()} will throw.
+     */
+    public abstract void setCursorError( String message );
```

Comment:
wdyt about renaming `setCursorError` and `clearCursorError` to `setCursorException` and `clearCursorException` while i'm reading `error` i'm really expection Error to be used for that then instead of exception and it's a bit confusing.